.github/workflows/msrv-rust-toolchain.toml

```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.61.0"
+channel = "1.63.0"
```

Comment:
I think this file doesn't end up being used since we switched to dtolnay/rust-toolchain. This can probably be removed now

Reply:
Ah actually it looks like we don't use stable rust for the stable CI check: msrv and stable checks are running the exact same thing. In the `check` job, we could use `matrix: [stable, 1.63]` and then:

```yml
- uses: dtolnay/rust-toolchain@master
  with:
    toolchain: ${{ matrix.toolchain }}
```

to replace the `helix-editor/toolchain` action.

Reply:
I am not sure we actually want to have a stable build.
Whenever a new rust version releases you get new clippy lints which means that PR piplines could just randomly start failing. This is especially problematic as clippy often inrdocues new lints which can only be fixed with new language features (example `then_some`). 

It would also be annoying development wise because the rust_toolchain file ist pinned to the `MSRV` so we only know about clippy lints once the CI fails. I think it honestly makes more sense to only run CI on `MSRV`.

Rust breaking backwars incomptatbility is rare enough that we don't need to worry about it and I would rather fix new clippy lints when the MSRV is bumped like in this PR so I don't see much value in running a stable CI. This would also speedup CI times which is not super important but a nice bonus

Reply:
Ah good point, I agree. I can't find the history for it but if I remember correctly, we pinned to 1.61 because we had a bunch of CI failures from a new rust version release being used automatically that ❌'d a bunch of open PR's CI https://github.com/helix-editor/helix/commit/aa87adf54b0b8dec07baa257aed71d09c3efb119. I would be in favor of just removing the "stable" branch from CI